### PR TITLE
[Ide] Run all WelcomePageService operations on main UI thread

### PR DESCRIPTION
**Pull request by sevoku**
Fixes VSTS #957299

### CODE REVIEW

**Comment by sevoku** [comment]
@monojenkins backport release-8.2

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.WelcomePage/WelcomePageService.cs [hunk 1]
```diff
@@ -102,60 +102,64 @@ internal static async Task Initialize (bool hideWelcomePage)
 
 		public static bool HasWindowImplementation => WelcomeWindowProvider != null;
 
-		public static async void ShowWelcomePageOrWindow (WelcomeWindowShowOptions options = null)
+		public static async Task ShowWelcomePageOrWindow (WelcomeWindowShowOptions options = null)
 		{
```
Comment (mrward) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.WelcomePage/WelcomePageService.cs]: Public API break? The WelcomePageService class is public so these Task changes are changing the public API.

Comment (sevoku) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.WelcomePage/WelcomePageService.cs]: Yes, that was intended, we have changed that several times already, this should be the final thing here. But I actually wanted to split the PR, will do that asap.

Comment (sevoku) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.WelcomePage/WelcomePageService.cs]: moved to #8345

**Comment by mrward** [comment]
The ShowWelcomePage and HideWelcomePage methods are public and could be run from a background thread. Should we add a Runtime.AssertMainThread in those methods?

Should the events fired here be fired only on the UI thread? It looks like WelcomeWindowShown and WelcomeWindowHidden could be fired on a background thread. For IDE events we normally only fire those on the UI thread.

**Comment by sevoku** [comment]
done @mrward 👍

**Comment by sevoku** [comment]
@monojenkins backport release-8.2

**Comment by sevoku** [comment]
@monojenkins backport release-8.3

**Comment by sevoku** [comment]
@monojenkins backport release-8.3
@monojenkins backport release-8.2

**Pull request merged**